packages/core/src/shipping/strategies/braintree/braintree-accelerated-checkout-shipping-strategy.ts

```diff
@@ -0,0 +1,221 @@
+        }
+
+        try {
+            await this._store.dispatch(
+                this._paymentMethodActionCreator.loadPaymentMethod(methodId),
+            );
+            await this._runAuthenticationFlowOrThrow(methodId);
+        } catch (error) {
+            // Info: we should not throw any error here to avoid
+            // customer stuck on shipping step due to the payment provider
+            // custom flow
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _shouldRunAuthenticationFlow(): boolean {
+        const state = this._store.getState();
+        const cartId = state.cart.getCart()?.id;
+        const payPalConnectSessionId = this._browserStorage.getItem('sessionId');
+        const paymentProviderCustomer = state.paymentProviderCustomer.getPaymentProviderCustomer();
+
+        return !paymentProviderCustomer?.authenticationState && payPalConnectSessionId === cartId;
+    }
+
+    private async _runAuthenticationFlowOrThrow(methodId: string): Promise<void> {
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const countries = state.countries.getCountries() || [];
+        const customer = state.customer.getCustomer();
+        const billingAddress = state.billingAddress.getBillingAddress();
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const { clientToken, initializationData } = paymentMethod;
+
+        if (!clientToken || !initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this._braintreeIntegrationService.initialize(clientToken, initializationData);
+
+        const braintreeConnect = await this._braintreeIntegrationService.getBraintreeConnect(
+            cart?.id,
+        );
+
+        const customerEmail = customer?.email || billingAddress?.email;
+
+        if (!customerEmail) {
+            return;
+        }
+
+        const { lookupCustomerByEmail, triggerAuthenticationFlow } = braintreeConnect.identity;
+
+        const { customerContextId } = await lookupCustomerByEmail(customerEmail);
+
+        if (!customerContextId) {
+            return;
+        }
+
+        const { authenticationState, profileData } = await triggerAuthenticationFlow(
+            customerContextId,
+        );
+
+        const addresses = this._mapPayPalToBcAddress(profileData.addresses, countries) || [];
+        const instruments = this._mapPayPalToBcInstrument(methodId, profileData.cards) || [];
+
+        await this._store.dispatch(
+            this._paymentProviderCustomerActionCreator.updatePaymentProviderCustomer({
+                authenticationState,
+                addresses,
+                instruments,
+            }),
+        );
+
+        if (addresses.length > 0) {
+            await this._store.dispatch(
+                this._billingAddressActionCreator.updateAddress({
```

Comment:
I see we are updating billing address here already, what happens for the case when this setting is off?
<img width="1039" alt="Screenshot 2023-08-31 at 10 35 45 pm" src="https://github.com/bigcommerce/checkout-sdk-js/assets/7134802/89539909-765a-4851-848d-19a9df4ce141">

Reply:
In this case we are also preset billing address as it is AXO flow, but after this user can disable this check mark and select address which he wants, as in general checkout flaw